BedrockServer.cpp

```diff
@@ -533,24 +533,24 @@ void BedrockServer::sync(const SData& args,
                 continue;
             }
 
-            // We want to run through all of the commands in our queue. However, we set a maximum limit. There's a race
-            // condition that can happen when leader is standing down. In that case, when we escalate a command,
-            // SQLiteNode immediately gives it back to the BedrockServer to handle, which will peek and then process it
-            // normally, and it can then be ready to be escalated again. So it gets back here, and this causes a loop
-            // of re-processing the command multiple times until the leader finishes standing down.
-            //
-            // This means that it's possible we get stuck in an infinite loop here until the server finishes standing
-            // down, though that in itself wouldn't be too horrible. It'd resolve when leader finished standing down.
-            // 
-            // There are two problems.
-            // 1. We don't know that leader is done standing down until we poll the sync node, so we need to get out of
-            // this loop to check.
-            // 2. The pipe we use inside of `syncNodeQueuedCommands` has a buffer size of 64k. If we add more than 64k
-            // items to the queue without reading from the pipe, we'll block further writes, causing deadlocks until we
-            // read again, but again, the read is done in postPoll, so we need to break out of this loop so we can
-            // clear that buffer.
-            // TODO: We could probably make writes non-blocking as well, and mitigate case 2 above, but we'd still have
-            // to deal with case 1.
+            // Don't escalate, leader can't handle the command anyway. Don't even dequeue the command, just leave it
+            // until one of these states changes. This prevents an endless loop of escalating commands, having
+            // SQLiteNode re-queue them because leader is standing down, and then escalating them again until leader
+            // sorts itself out.
+            if (nodeState == SQLiteNode::FOLLOWING && server._syncNode->leaderState() == SQLiteNode::STANDINGDOWN) {
```

Comment:
Wouldn't it be better to do `!= SQLiteNode::LEADING`?

Reply:
The state can only be LEADING, FOLLOWING, or STANDINGDOWN here.

Reply:
Sorry, I meant the leaderState, which can be anything at this point right?

Reply:
I think they're equivalent, the leader can only be LEADING or STANDINGDOWN, otherwise it's not the leader.

Reply:
Got it, we will only have a `_leadPeer` if we had a host  in STANDINGUP, STANDINGDOWN, or LEADING last time we set it.

Reply:
We won't have a `_leadPeer` in `STANDINGUP`, it will only be set in `LEADING`, and will be cleared if anything strange happens (the only valid transition out of `LEADING` is to `STANDINGDOWN`).